Terminal.Gui/Drawing/ColorScheme.cs

```diff
@@ -4,249 +4,210 @@
-												EqualityComparer<Attribute>.Default.Equals (_disabled, other._disabled);
-
-	/// <summary>
-	/// Compares two <see cref="ColorScheme"/> objects for equality.
-	/// </summary>
-	/// <param name="obj"></param>
-	/// <returns>true if the two objects are equal</returns>
-	public override bool Equals (object? obj) => Equals (obj as ColorScheme);
-
-	/// <summary>
-	/// Returns a hashcode for this instance.
-	/// </summary>
-	/// <returns>hashcode for this instance</returns>
-	public override int GetHashCode ()
-	{
-		var hashCode = -1242460230;
-		hashCode = hashCode * -1521134295 + _normal.GetHashCode ();
-		hashCode = hashCode * -1521134295 + _focus.GetHashCode ();
-		hashCode = hashCode * -1521134295 + _hotNormal.GetHashCode ();
-		hashCode = hashCode * -1521134295 + _hotFocus.GetHashCode ();
-		hashCode = hashCode * -1521134295 + _disabled.GetHashCode ();
-		return hashCode;
-	}
-
-	/// <summary>
-	/// Compares two <see cref="ColorScheme"/> objects for equality.
-	/// </summary>
-	/// <param name="left"></param>
-	/// <param name="right"></param>
-	/// <returns><c>true</c> if the two objects are equivalent</returns>
-	public static bool operator == (ColorScheme left, ColorScheme right) => EqualityComparer<ColorScheme>.Default.Equals (left, right);
-
-	/// <summary>
-	/// Compares two <see cref="ColorScheme"/> objects for inequality.
-	/// </summary>
-	/// <param name="left"></param>
-	/// <param name="right"></param>
-	/// <returns><c>true</c> if the two objects are not equivalent</returns>
-	public static bool operator != (ColorScheme left, ColorScheme right) => !(left == right);
-
-	/// <inheritdoc />
-	public override string ToString ( ) => $"Normal: {Normal}; Focus: {Focus}; HotNormal: {HotNormal}; HotFocus: {HotFocus}; Disabled: {Disabled}";
+    /// <summary>Creates a new instance set to the default colors (see <see cref="Attribute.Default"/>).</summary>
+    public ColorScheme () : this (Attribute.Default) { }
+
+    /// <summary>Creates a new instance, initialized with the values from <paramref name="scheme"/>.</summary>
+    /// <param name="scheme">The scheme to initialize the new instance with.</param>
+    public ColorScheme (ColorScheme scheme) {
+        if (scheme is null) {
+            throw new ArgumentNullException (nameof (scheme));
+        }
+
+        _normal = scheme.Normal;
+        _focus = scheme.Focus;
+        _hotNormal = scheme.HotNormal;
+        _disabled = scheme.Disabled;
+        _hotFocus = scheme.HotFocus;
+    }
+
+    /// <summary>Creates a new instance, initialized with the values from <paramref name="attribute"/>.</summary>
+    /// <param name="attribute">The attribute to initialize the new instance with.</param>
+    public ColorScheme (Attribute attribute) {
+        _normal = attribute;
+        _focus = attribute;
+        _hotNormal = attribute;
+        _disabled = attribute;
+        _hotFocus = attribute;
+    }
+
+    private readonly Attribute _disabled;
```

Comment:
Are the layout rules correct here?

Don't we want private fields to be first or, if they back a property, directly above their associated properties?

Reply:
Can you tell what setting is in correct. I thought I had this right.

Reply:
That'll be in the file layout configuration. There's also an extension that adds even more capabilities to that functionality.

I was pretty sure we'd end up with at least a couple things not being exactly as we wanted, and this part was likely to be one of them.

That stuff is powerful. But, with that power, comes a fair amount of complexity in getting it right. Then you pretty much never have to touch it again until the language changes enough to matter.